src/components/Calculator.js

```diff
@@ -1,46 +1,32 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Calculator.css';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-    this.handleClick = this.handleClick.bind(this);
-  }
+function Calculator() {
+  const [calcObject, setCalcObject] = useState({});
 
-  handleClick(e) {
-    this.setState((state) => calculate(state, e.target.textContent));
+  function handleClick(e) {
+    setCalcObject({ ...calcObject, ...calculate(calcObject, e.target.textContent) });
```

Comment:
- [ ] I think you need to implement error handling method or condition. When an operator button is clicked twice or repeatedly it throws an error on the browser. Kindly fix this by implementing error handling. :pray:

![Screenshot from 2021-08-20 15-41-15](https://user-images.githubusercontent.com/64894927/130214640-eb4caccb-3fe8-4657-9e13-838c8f5bb543.png)

Reply:
error handling was implemented.